src/onnx/parse_dynamicquantizelinear.cpp

```diff
@@ -0,0 +1,152 @@
+* where qmax and qmin are max and min values for quantization range i.e. [0, 255] in case of uint8
+* data range is adjusted to include 0.
+
+Zero point is calculated as:
+intermediate_zero_point = qmin - min(x)/y_scale
+y_zero_point = cast(round(saturate(itermediate_zero_point)))
+* where qmax and qmin are max and min values for quantization range .i.e [0, 255] in case of uint8
+* for saturation, it saturates to [0, 255] if it's uint8, or [-127, 127] if it's int8. Right now
+only uint8 is supported.
+* rounding to nearest ties to even. Data quantization formula is:
+
+y = saturate (round (x / y_scale) + y_zero_point)
+* for saturation, it saturates to [0, 255] if it's uint8, or [-127, 127] if it's int8.Right now only
+uint8 is supported.
+* rounding to nearest ties to even.
+
+Version
+This version of the operator has been available since version 11 of the default ONNX operator set.
+
+Inputs
+x : T1
+Input tensor
+
+Outputs
+y : T2
+Quantized output tensor
+
+y_scale : tensor(float)
+Output scale. It's a scalar, which means a per-tensor/layer quantization.
+
+y_zero_point : T2
+Output zero point. It's a scalar, which means a per-tensor/layer quantization.
+
+Type Constraints
+T1 : tensor(float)
+Constrain 'x' to float tensor.
+
+T2 : tensor(uint8)
+Constrain 'y_zero_point' and 'y' to 8-bit unsigned integer tensor.
+*/
+
+struct parse_dynamicquantizelinear : op_parser<parse_dynamicquantizelinear>
+{
+    std::vector<op_desc> operators() const { return {{"DynamicQuantizeLinear"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd*/,
+                                       const onnx_parser& /*parser*/,
+                                       const onnx_parser::node_info& info,
+                                       const std::vector<instruction_ref>& args) const
+    {
+        auto x       = args[0];
+        auto x_shape = x->get_shape();
+        if(x_shape.dynamic())
+            MIGRAPHX_THROW("DYNAMICQUANTIZELINEAR: dynamic shapes are not supported");
+
+        auto x_type = x_shape.type();
+        // 1. Computing y_scale
+        auto l0 = info.add_literal({0.f});
+        // DynamicQuantizeLinear only has uint8 quantization
+        auto q_max = info.add_literal(
+            migraphx::literal{migraphx::shape{x_type}, {std::numeric_limits<uint8_t>::max()}});
+        auto q_min = info.add_literal(
+            migraphx::literal{migraphx::shape{x_type}, {std::numeric_limits<uint8_t>::min()}});
+        auto x_reshape = x;
+        if(x_shape.lens().size() != 1)
+        {
+            x_reshape = info.add_instruction(
+                migraphx::make_op("reshape", {{"dims", {x_shape.elements()}}}), x);
+        }
+        x_reshape = info.add_instruction(migraphx::make_op("concat", {{"axis", 0}}), x_reshape, l0);
+        // maximum(0, max(x))
+        auto topk_max = info.add_instruction(
+            migraphx::make_op("topk", {{"axis", 0}, {"k", 1}, {"largest", true}}), x_reshape);
```

Comment:
Is it preferable to use `topk` over the equivalent of `np.max(x)`? I believe we currently parse ONNX's `Max` operator into multiple MIGX `max` operators. Probably a question for @pfultz2.

Reply:
> Is it preferable to use topk over the equivalent of np.max(x)? I

`np.max(x)` does sound like a better option. 

> I believe we currently parse ONNX's Max operator into multiple MIGX max operators.

Yea for when its variadic.

Reply:
> I also looked at the choices, and [Onnx Max](https://xadupre.github.io/draft/onnx/onnx_doc_folder/onnx__Max.html) wasn't the right option. We need some kind of equivalent of std::max() & std::min(), if there is any.

Charlie is suggesting to use something equivalent to `np.max`, which would be `reduce_max` in migraphx(or [`ReduceMax`](https://xadupre.github.io/draft/onnx/onnx_doc_folder/onnx__ReduceMax.html) in onnx).

Our [`max`](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/op/max.hpp#L45) and [`min`](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/op/min.hpp#L45) operator already uses `std::max` and `std::min`, but this is applied per element on the tensor(similar to `np.maximum` not `np.max`).